airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -159,7 +164,16 @@ private static Container getInit(final boolean usesStdin,
       initEntrypointStr = String.format("mkfifo %s && ", STDIN_PIPE_FILE) + initEntrypointStr;
     }
 
-    initEntrypointStr = initEntrypointStr + String.format(" && until [ -f %s ]; do sleep 0.1; done;", SUCCESS_FILE_NAME);
+    initEntrypointStr = initEntrypointStr +
```

Comment:
is there any sane way to heartbeat instead? 

timeouts could be such a headache here. if we do need a time out, are we confident that in 99% of cases that if it is taking longer than 5 min it is that something is broken and that it isn't a big config?

also, if we ever do exit due to timeout, I think it is worth explicitly logging that the error was due to a timeout to make it easier to debug.

Reply:
I'm looking through how the heartbeat server works now, I just wanted to put this PR up to build a version of the timeout because it was pretty low effort to throw together. 5 minutes is arbitrary but yeah, you make a good point that no matter how high we set it, there'd always be some concern that we just have a massive config to copy over and we end up killing a working container.

Reply:
A kube cluster internal `kubectl cp` should be moving data at >1MB/s. Are we expecting 300mb+ configs?

Reply:
Another way we could approach this is to check to see if the disk usage in the folder is increasing over time. If it's increasing, we could keep on resetting our timeout, but if it's stalled and not increasing for 1min that seems safe. 

If we do want to allow this to use heartbeating, I think the approach would be to get rid of the init container entirely and instead move this waiting logic to the main container. Then we'd also need to change the `KubePodProcess` logic to copy files into the main container instead of the init container, and change the watches for pod status to look at the other containers.

For the orchestrator, we don't want a consistent connection between the two so we either need the heartbeat as it exists today or some additional check that the transfer is actually occurring.

Reply:
Also, fwiw, I'm fine with going with the timeout now (as long as it logs when the timeout is the thing that terminates it) and create a follow up issue to be smarter later.